### ISSUE TITLE
Extra artwork

### ISSUE THREAD

**Issue by Montellese** [Montellese/mediaimporter.plex#18]
Does Plex support extra artwork (landscape, clearart, logo, discart, ...)?

**Comment by JeordyR** [comment]
It doesn't look like plex stores any extra artwork at all.

The only image values I see are:
`thumb="/library/metadata/48519/thumb/1542616338"`
and
`art="/library/metadata/48519/art/1542616338"`
for the actual movie, then thumb values for the actors; similar for TV Shows.

Since plex does have the tmdbid (and tvdbid for TV show stuff) we could potentially parse those then pull extra artwork from them. Not sure if you want to deal with hitting external sources within this addon or not.

**Comment by Montellese** [comment]
I'm not a fan of doing such things in an add-on. Kodi would be perfectly capable of looking up additional artwork.

In my initial design of media import I always considered the case that it might make sense to lookup / parse additional information if a media provider cannot provide specific information (in this case extra artwork).

But I never had a real case where this was needed so far which is why I never integrated it.

Maybe we should allow media importers to provide a list of information which should be retrieved using Kodi's scrapers instead or something like that. Auto-detecting whether some information is missing because it also doesn't exist in the online databases or because the provider doesn't provide it would be very difficult.

**Comment by JeordyR** [comment]
> I'm not a fan of doing such things in an add-on. Kodi would be perfectly capable of looking up additional artwork.
> 
> In my initial design of media import I always considered the case that it might make sense to lookup / parse additional information if a media provider cannot provide specific information (in this case extra artwork).
> 
> But I never had a real case where this was needed so far which is why I never integrated it.
> 
> Maybe we should allow media importers to provide a list of information which should be retrieved using Kodi's scrapers instead or something like that. Auto-detecting whether some information is missing because it also doesn't exist in the online databases or because the provider doesn't provide it would be very difficult.

I agree, seems unnecessary to put the extra burden of Kodi-specific metadata on a media importer. Would skins not trigger the scraping of any extra metadata they want but don't currently have? Seems like having the media importers provide a list of things they don't have makes them have to be unnecessarily aware of/bound to Kodi's metadata needs as well vs just giving what they have and letting Kodi figure out the rest. Just not sure what "Kodi figure out the rest" looks like or what options there are.
